importccl: avoid TypeCheck erroring the import if column does not exist

otan opened the pull request:
Refs: #52382 

Since we now do type checks checking for AddGeometryColumn during IMPORT,
columns which do not exist during the type check (as they are not created
yet) will fail out the entire import. Avoid this by checking explicitly for
the column not exists error during type check, and if that is the case,
ignore processing the SELECT statement during IMPORT time.

This is a specialization of a case in ogr2ogr, where the sequence maxval
is set to `max("fid")`. Our existing code will ignore this set up
(as we do not by default support making sequences for SERIAL columns),
so this feels safe to be a class of error to ignore.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52502)
<!-- Reviewable:end -->

otan committed:
importccl: avoid TypeCheck erroring the import if column does not exist

Since we now do type checks checking for AddGeometryColumn during IMPORT,
columns which do not exist during the type check (as they are not created
yet) will fail out the entire import. Avoid this by checking explicitly for
the column not exists error during type check, and if that is the case,
ignore processing the SELECT statement during IMPORT time.

This is a specialization of a case in ogr2ogr, where the sequence maxval
is set to `max("fid")`. Our existing code will ignore this set up
(as we do not by default support making sequences for SERIAL columns),
so this feels safe to be a class of error to ignore.

Release note: None

otan commented:
TFTRs!

bors r=mjibson,pbardea
